assets/js/base/components/product-grid-item/product-price/index.js

```diff
@@ -4,45 +4,62 @@
 import PropTypes from 'prop-types';
 import NumberFormat from 'react-number-format';
 
-const ProductPrice = ( { className, product = {} } ) => {
+const ProductPrice = ( { className, prices } ) => {
 	const numberFormatArgs = {
 		displayType: 'text',
-		thousandSeparator: product.prices.thousand_separator,
-		decimalSeparator: product.prices.decimal_separator,
-		decimalScale: product.prices.decimals,
-		prefix: product.prices.price_prefix,
-		suffix: product.prices.price_suffix,
+		thousandSeparator: prices.thousand_separator,
+		decimalSeparator: prices.decimal_separator,
+		decimalScale: prices.decimals,
+		prefix: prices.price_prefix,
+		suffix: prices.price_suffix,
 	};
 
-	if ( product.prices.price_range && product.prices.price_range.min_amount && product.prices.price_range.max_amount ) {
+	if (
+		prices.price_range &&
+		prices.price_range.min_amount &&
+		prices.price_range.max_amount
+	) {
 		return (
 			<div className={ className }>
 				<span className={ className + '__value' }>
-					<NumberFormat value={ product.prices.price_range.min_amount } { ...numberFormatArgs } />
+					<NumberFormat
+						value={ prices.price_range.min_amount }
+						{ ...numberFormatArgs }
+					/>
 					&nbsp;&mdash;&nbsp;
-					<NumberFormat value={ product.prices.price_range.max_amount } { ...numberFormatArgs } />
+					<NumberFormat
+						value={ prices.price_range.max_amount }
+						{ ...numberFormatArgs }
+					/>
 				</span>
 			</div>
 		);
 	}
 
 	return (
 		<div className={ className }>
-			{ product.prices.regular_price !== product.prices.price && (
+			{ prices.regular_price && prices.regular_price !== prices.price && (
```

Comment:
![Screenshot 2019-09-30 at 14 35 16](https://user-images.githubusercontent.com/90977/65883746-8e601200-e38f-11e9-8b3e-35539fe648b7.png)

I'm getting a stray 0 here for some reason.

Reply:
Is that happening only with this product or all of them? I'm unable to reproduce.

Reply:
Only this one which has no cost.

Reply:
Right, good catch. Fixed in c0a6670dc1046c90accd98294ab21c0f3e384744.